File: Unstop/Practice Problems/Medium/happynumber.py
def sum_of_squares_of_digits(n):
    # Function to calculate the sum of squares of digits of a number
    total = 0
    while n > 0:
        digit = n % 10
        total += digit * digit
        n //= 10
    return total

def is_good_number(n):
    slow = n
    fast = n

    while True:
        slow = sum_of_squares_of_digits(slow)
        fast = sum_of_squares_of_digits(sum_of_squares_of_digits(fast))

        if fast == 1:
            return True
        if slow == fast:
            return False

# Input number
n = int(input("Enter the number: "))

# Check if it's a good number and print the result
print(is_good_number(n))

File: Unstop/Practice Problems/Medium/HappyNumber.java
import java.util.Scanner;

public class HappyNumber {
    public static int sumOfSquaresOfDigits(int n) {
        int sum = 0;
        while (n > 0) {
            int digit = n % 10;
            sum += digit * digit;
            n /= 10;
        }
        return sum;
    }

    public static boolean isGoodNumber(int n) {
        int slow = n, fast = n;

        do {
            slow = sumOfSquaresOfDigits(slow);
            fast = sumOfSquaresOfDigits(sumOfSquaresOfDigits(fast));

            if (fast == 1)
                return true;

        } while (slow != fast);

        return false;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number: ");
        int n = scanner.nextInt();
        scanner.close();

        System.out.println(isGoodNumber(n));
    }
}
